tests/factories.py

```diff
@@ -128,11 +141,37 @@ class Meta:
     return hosts[0], hosts[1]
 
 
-# async def connection_pair_factory() -> Tuple[Mplex, BasicHost, Mplex, BasicHost]:
-#     host_0, host_1 = await host_pair_factory()
-#     mplex_conn_0 = host_0.get_network().connections[host_1.get_id()]
-#     mplex_conn_1 = host_1.get_network().connections[host_0.get_id()]
-#     return mplex_conn_0, host_0, mplex_conn_1, host_1
+async def swarm_conn_pair_factory(
+    is_secure: bool, muxer_opt: TMuxerOptions = None
+) -> Tuple[SwarmConn, Swarm, SwarmConn, Swarm]:
+    swarms = await swarm_pair_factory(is_secure)
+    conn_0 = swarms[0].connections[swarms[1].get_peer_id()]
+    conn_1 = swarms[1].connections[swarms[0].get_peer_id()]
+    return conn_0, swarms[0], conn_1, swarms[1]
+
+
+async def mplex_conn_pair_factory(is_secure: bool) -> Tuple[Mplex, Swarm, Mplex, Swarm]:
+    muxer_opt = {MPLEX_PROTOCOL_ID: Mplex}
+    conn_0, swarm_0, conn_1, swarm_1 = await swarm_conn_pair_factory(
+        is_secure, muxer_opt=muxer_opt
+    )
+    return conn_0.muxed_conn, swarm_0, conn_1.muxed_conn, swarm_1
+
+
+async def mplex_stream_pair_factory(
+    is_secure: bool
+) -> Tuple[MplexStream, Swarm, MplexStream, Swarm]:
+    mplex_conn_0, swarm_0, mplex_conn_1, swarm_1 = await mplex_conn_pair_factory(
+        is_secure
+    )
+    stream_0 = await mplex_conn_0.open_stream()
+    await asyncio.sleep(0.01)
```

Comment:
do we need this sleep here?

Reply:
Yes I think so. if we want to be more confident that the `mplex_conn_0` receives the `NewStream` message and accept the `MplexStream`.